### PR TITLE
HPCC-18691 Do not challenge OPTIONS requests

### PR DESCRIPTION
**Pull request by rpastrana**
- Processes CORS OPTIONS requests without athorization

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by rpastrana** [comment]
@afishbeck please take a look. This resolves the tableau desktop communication error. We can discuss if we feel this opens a security issue.

**Comment by richardkchapman** [comment]
Typo in commit message.

File: esp/bindings/http/platform/httpservice.cpp [hunk 1]
```diff
@@ -259,6 +259,9 @@ int CEspHttpServer::processRequest()
         const char *userid=ctx->queryUserId();
         ESPLOG(LogMin, "%s %s, from %s@%s", method.str(), m_request->getPath(pathStr).str(), (userid) ? userid : "unknown", m_request->getPeer(peerStr).str());
 
+        if(strieq(method.str(), OPTIONS_METHOD))
+            return onOptions();
```
Comment (afishbeck) [inline esp/bindings/http/platform/httpservice.cpp]: Should probably delete the identical lines below?

**Comment by afishbeck** [comment]
@rpastrana looks good except duplicate code.  But I wonder if this was always like this, if so this change should probably be 6.4.6.  Unless it was broken by the session changes that are targeting 7.0.

**Comment by rpastrana** [comment]
new pull request targets earlier branch: 
https://github.com/hpcc-systems/HPCC-Platform/pull/10628

**Pull request closed without merging**